topdown/http.go

```diff
@@ -462,8 +498,18 @@ func createHTTPRequest(bctx BuiltinContext, obj ast.Object) (*http.Request, *htt
 	}
 
 	if isTLS {
-		client.Transport = &http.Transport{
-			TLSClientConfig: &tlsConfig,
+		if ok, parsedURL, tr := useSocket(url, &tlsConfig); ok {
+			client.Transport = tr
+			url = parsedURL
+		} else {
+			client.Transport = &http.Transport{
```

Comment:
🔍 here's another zero-value-riddled `http.Transport`. But I guess this is old code, we should perhaps audit the builtin's code for these. (Outside of this PR.)